satpy/readers/abi_base.py

```diff
@@ -0,0 +1,220 @@
+                                      chunks={'x': CHUNK_SIZE, 'y': CHUNK_SIZE},)
+        except ValueError:
+            self.nc = xr.open_dataset(self.filename,
+                                      decode_cf=True,
+                                      mask_and_scale=False,
+                                      chunks={'lon': CHUNK_SIZE, 'lat': CHUNK_SIZE},)
+
+        self.nc = self.nc.rename({'t': 'time'})
+        platform_shortname = filename_info['platform_shortname']
+        self.platform_name = PLATFORM_NAMES.get(platform_shortname)
+        self.sensor = 'abi'
+
+        if 'goes_imager_projection' in self.nc.keys():
+            self.nlines = self.nc['y'].size
+            self.ncols = self.nc['x'].size
+        elif 'goes_lat_lon_projection' in self.nc.keys():
+            self.nlines = self.nc['lat'].size
+            self.ncols = self.nc['lon'].size
+            self.nc = self.nc.rename({'lon': 'x', 'lat': 'y'})
+
+        self.coords = {}
+
+    def __getitem__(self, item):
+        """Wrapper around `self.nc[item]`.
+
+        Some datasets use a 32-bit float scaling factor like the 'x' and 'y'
+        variables which causes inaccurate unscaled data values. This method
+        forces the scale factor to a 64-bit float first.
+
+        """
+        data = self.nc[item]
+        attrs = data.attrs
+        factor = data.attrs.get('scale_factor')
+        offset = data.attrs.get('add_offset')
+        fill = data.attrs.get('_FillValue')
+        if fill is not None:
+            data = data.where(data != fill)
+        if factor is not None:
+            # make sure the factor is a 64-bit float
+            # can't do this in place since data is most likely uint16
+            # and we are making it a 64-bit float
+            data = data * float(factor) + offset
+        data.attrs = attrs
+
+        # handle coordinates (and recursive fun)
+        new_coords = {}
+        # 'time' dimension causes issues in other processing
+        # 'x_image' and 'y_image' are confusing to some users and unnecessary
+        # 'x' and 'y' will be overwritten by base class AreaDefinition
+        for coord_name in ('x_image', 'y_image', 'time', 'x', 'y'):
+            if coord_name in data.coords:
+                del data.coords[coord_name]
+        if item in data.coords:
+            self.coords[item] = data
+        for coord_name in data.coords.keys():
+            if coord_name not in self.coords:
+                self.coords[coord_name] = self[coord_name]
+            new_coords[coord_name] = self.coords[coord_name]
+        data.coords.update(new_coords)
+
+        return data
+
+    def get_dataset(self, key, info):
+        """Load a dataset."""
+        raise NotImplementedError("Reader {} has not implemented get_dataset".format(self.name))
+
+    def get_area_def(self, key):
+        """Get the area definition of the data at hand.
+        """
+        if 'goes_imager_projection' in self.nc.keys():
```

Comment:
If I'm understanding this correctly you are checking if the 'goes_imager_projection' variable exists. If so, you should change this to `'goes_imager_projection' in self.nc`. If that doesn't work use `self.nc.variables`. By doing `self.nc.keys()` you are creating an iterator in Python 3 and requiring it to be iterated until it finds the key. If you do `self.nc` or `self.nc.variables` then it will be using dictionary lookup which will be much faster.

Reply:
@djhoese Thanks for reviewing, I have updated AUTHERS.md and the copyright head, pop the '_Unsigned' and the self.nc works well too.